File: integration-tests/iac/lambdas/manual/handler.js
'use strict';

const { meter } = require('./tracing');

// Create a counter metric
const invocationCounter = meter.createCounter('lambda.invocations', {
  description: 'Number of Lambda invocations',
});

module.exports.hello = async (event, context) => {
  // Record a metric datapoint
  invocationCounter.add(1, {
    'function.name': context?.functionName || 'unknown',
  });

  return {
    statusCode: 200,
    body: JSON.stringify({ message: "Success" }),
  };
};

File: integration-tests/iac/lambdas/python/dependency-conflict/handler.py
from google.protobuf import __version__ as protobuf_version
from google.protobuf.struct_pb2 import Struct


def handler(event, context):
    print(f"protobuf version: {protobuf_version}")

    # Create a simple protobuf Struct
    s = Struct()
    s.update({"message": "Hello from protobuf!", "number": 42})

    return {
        "statusCode": 200,
        "body": f"protobuf version: {protobuf_version}, struct: {s}"
    }
